packages/cli/internal/pkg/cli/account_activate.go

```diff
@@ -61,62 +65,97 @@ func newAccountActivateOpts(vars accountActivateVars) (*accountActivateOpts, err
 		stsClient:           aws.StsClient(profile),
 		s3Client:            aws.S3Client(profile),
 		cdkClient:           cdk.NewClient(profile),
+		cfnClient:           aws.CfnClient(profile),
 		region:              aws.Region(profile),
 	}, nil
 }
 
 // Execute activates AGC.
 func (o *accountActivateOpts) Execute() error {
+
+	environmentVars, err := o.generateEnvVars()
+	if err != nil {
+		return err
+	}
+
+	homeDir, err := osutils.DetermineHomeDir()
+	if err != nil {
+		return err
+	}
+
+	cdkAppPath := filepath.Join(homeDir, cdkCoreDir)
+	err = o.cdkBootstrap(cdkAppPath, environmentVars)
+	if err != nil {
+		return err
+	}
+
+	return o.deployCoreInfrastructure(cdkAppPath, environmentVars)
+}
+
+func (o accountActivateOpts) generateDefaultBucket() (string, error) {
+	account, err := o.stsClient.GetAccount()
+	if err != nil {
+		return "", err
+	}
+	return generateBucketName(account, o.region), nil
+}
+
+func (o accountActivateOpts) generateEnvVars() ([]string, error) {
```

Comment:
I understand that you've created this function as `Execute()` is getting too big. Although you're catching the err on line 97, the err handling flow has changed and the program now wait until all these checks are complete as opposed to original fail-fast which is a better mechanism in my opinion. 

Let me know if I'm missing something.

Reply:
Hey Angela,
If you look through the function, we exit out of the fail whenever the function returns an error so logically the code is exactly the same but it is easier to read. 
I.e. the error from line 97 is passed to line 108 which is then passed to line 79. The original flow would have been 97 to line 79.

Reply:
Thanks Elliot for the detail explanation. I read it more closely now. It is doing I was intended. No change is required here.